File: Golang/Chapter 5/Sending transactions from high-load wallet.go
package main

import (
	"context"
	"crypto/ed25519"
	"crypto/hmac"
	"crypto/sha512"
	"fmt"
	"github.com/xssnick/tonutils-go/address"
	"github.com/xssnick/tonutils-go/liteclient"
	"github.com/xssnick/tonutils-go/tl"
	"github.com/xssnick/tonutils-go/tlb"
	"github.com/xssnick/tonutils-go/ton"
	"github.com/xssnick/tonutils-go/tvm/cell"
	"golang.org/x/crypto/pbkdf2"
	"log"
	"math/big"
	"math/rand"
	"strings"
	"time"
)

func main() {
	var internalMessages []*cell.Cell
	wallletAddress := address.MustParseAddr("Eput your wallet address from which you deployed high-load wallet")

	for i := 0; i < 12; i++ {
		comment := fmt.Sprintf("Hello, TON! #%d", i)
		internalMessageBody := cell.BeginCell().
			MustStoreUInt(0, 32).
			MustStoreBinarySnake([]byte(comment)).
			EndCell()

		internalMessage := cell.BeginCell().
			MustStoreUInt(0x18, 6). // bounce
			MustStoreAddr(wallletAddress).
			MustStoreBigCoins(tlb.MustFromTON("0.001").NanoTON()).
			MustStoreUInt(0, 1+4+4+64+32).
			MustStoreBoolBit(false).           // We do not have State Init
			MustStoreBoolBit(true).            // We store Message Body as a reference
			MustStoreRef(internalMessageBody). // Store Message Body Init as a reference
			EndCell()

		messageData := cell.BeginCell().
			MustStoreUInt(3, 8). // transaction mode
			MustStoreRef(internalMessage).
			EndCell()

		internalMessages = append(internalMessages, messageData)
	}

	dictionary := cell.NewDict(16) // create an empty dictionary with the key as a number and the value as a cell
	for i := 0; i < len(internalMessages); i++ {
		internalMessage := internalMessages[i]                             // get our message from an array
		err := dictionary.SetIntKey(big.NewInt(int64(i)), internalMessage) // save the message in the dictionary
		if err != nil {
			return
		}
	}

	queryID := rand.Uint32()
	timeout := 120                                                               // timeout for message expiration, 120 seconds = 2 minutes
	now := time.Now().Add(time.Duration(timeout)*time.Second).UTC().Unix() << 32 // get current timestamp + timeout
	finalQueryID := uint64(now) + uint64(queryID)                                // get our final query_id
	log.Println(finalQueryID)                                                    // print query_id. With this query_id we can call GET method to check if our request has been processed

	toSign := cell.BeginCell().
		MustStoreUInt(698983191, 32). // subwallet_id
		MustStoreUInt(finalQueryID, 64).
		MustStoreDict(dictionary)

	highloadMnemonicArray := strings.Split("put your high-load wallet mnemonic", " ") // word1 word2 word3
	mac := hmac.New(sha512.New, []byte(strings.Join(highloadMnemonicArray, " ")))
	hash := mac.Sum(nil)
	k := pbkdf2.Key(hash, []byte("TON default seed"), 100000, 32, sha512.New) // In TON libraries "TON default seed" is used as salt when getting keys
	// 32 is a key len
	highloadPrivateKey := ed25519.NewKeyFromSeed(k) // get private key
	highloadWalletAddress := address.MustParseAddr("put your high-load wallet address")

	signature := ed25519.Sign(highloadPrivateKey, toSign.EndCell().Hash())

	body := cell.BeginCell().
		MustStoreSlice(signature, 512). // store signature
		MustStoreBuilder(toSign).       // store our message
		EndCell()

	externalMessage := cell.BeginCell().
		MustStoreUInt(0b10, 2).               // ext_in_msg_info$10
		MustStoreUInt(0, 2).                  // src -> addr_none
		MustStoreAddr(highloadWalletAddress). // Destination address
		MustStoreCoins(0).                    // Import Fee
		MustStoreBoolBit(false).              // No State Init
		MustStoreBoolBit(true).               // We store Message Body as a reference
		MustStoreRef(body).                   // Store Message Body as a reference
		EndCell()

	connection := liteclient.NewConnectionPool()
	configUrl := "https://ton-blockchain.github.io/global.config.json"
	err := connection.AddConnectionsFromConfigUrl(context.Background(), configUrl)
	if err != nil {
		panic(err)
	}
	client := ton.NewAPIClient(connection)

	var resp tl.Serializable
	err = client.Client().QueryLiteserver(context.Background(), ton.SendMessage{Body: externalMessage.ToBOCWithFlags(false)}, &resp)

	if err != nil {
		log.Fatalln(err.Error())
		return
	}
}

File: JavaScript - TypeScript/src/Chapter 5/Sending transactions from high-load wallet.ts
import { Address, Cell, Dictionary, beginCell, toNano } from "@ton/core";
import { mnemonicToWalletKey, sign } from "@ton/crypto";
import { TonClient } from "@ton/ton";
import * as crypto from 'crypto';

async function main() {
    let internalMessages:Cell[] = [];
    const walletAddress = Address.parse('put your wallet address from which you deployed high-load wallet');

    for (let i = 0; i < 12; i++) {
        const internalMessageBody = beginCell()
            .storeUint(0, 32)
            .storeStringTail(`Hello, TON! #${i}`)
            .endCell();

        const internalMessage = beginCell()
            .storeUint(0x18, 6) // bounce
            .storeAddress(walletAddress)
            .storeCoins(toNano('0.01'))
            .storeUint(0, 1 + 4 + 4 + 64 + 32)
            .storeBit(0) // We do not have State Init
            .storeBit(1) // We store Message Body as a reference
            .storeRef(internalMessageBody) // Store Message Body Init as a reference
            .endCell();

        internalMessages.push(internalMessage);
    }

    const dictionary = Dictionary.empty<number, Cell>(); // create an empty dictionary with the key as a number and the value as a cell
    for (let i = 0; i < internalMessages.length; i++) {
        const internalMessage = internalMessages[i]; // get our message from an array
        dictionary.set(i, internalMessage); // save the message in the dictionary
    }

    const queryID = crypto.randomBytes(4).readUint32BE(); // create a random uint32 number, 4 bytes = 32 bits
    const now = Math.floor(Date.now() / 1000); // get current timestamp
    const timeout = 120; // timeout for message expiration, 120 seconds = 2 minutes
    const finalQueryID = (BigInt(now + timeout) << 32n) + BigInt(queryID); // get our final query_id
    console.log(finalQueryID); // print query_id. With this query_id we can call GET method to check if our request has been processed

    const toSign = beginCell()
        .storeUint(698983191, 32) // subwallet_id
        .storeUint(finalQueryID, 64)
        // Here we create our own method that will save the 
        // transaction mode and a reference to the transaction
        .storeDict(dictionary, Dictionary.Keys.Int(16), {
            serialize: (src, buidler) => {
                buidler.storeUint(3, 8); // save transaction mode, mode = 3
                buidler.storeRef(src); // save transaction as reference
            },
            // We won't actually use this, but this method 
            // will help to read our dictionary that we saved
            parse: (src) => {
                let cell = beginCell()
                    .storeUint(src.loadUint(8), 8)
                    .storeRef(src.loadRef())
                    .endCell();
                return cell;
            }
        }
    );

    const highloadMnemonicArray = 'put your high-load wallet mnemonic'.split(' ');
    const highloadKeyPair = await mnemonicToWalletKey(highloadMnemonicArray); // extract private and public keys from mnemonic
    const highloadWalletAddress = Address.parse('put your high-load wallet address');

    const signature = sign(toSign.endCell().hash(), highloadKeyPair.secretKey); // get the hash of our message to wallet smart contract and sign it to get signature

    const body = beginCell()
        .storeBuffer(signature) // store signature
        .storeBuilder(toSign) // store our message
        .endCell();

    const externalMessage = beginCell()
        .storeUint(0b10, 2) // indicate that it is an incoming external transaction
        .storeUint(0, 2) // src -> addr_none
        .storeAddress(highloadWalletAddress)
        .storeCoins(0) // Import fee
        .storeBit(0) // We do not have State Init
        .storeBit(1) // We store Message Body as a reference
        .storeRef(body) // Store Message Body as a reference
        .endCell();

    // We do not need a key here as we will be sending 1 request per second
    const client = new TonClient({
        endpoint: 'https://toncenter.com/api/v2/jsonRPC',
        // apiKey: 'put your api key' // you can get an api key from @tonapibot bot in Telegram
    });

    client.sendFile(externalMessage.toBoc());
}

main().finally(() => console.log("Exiting..."));